source/deployment/cluster.rst

```diff
@@ -134,75 +133,147 @@ A sample configuration for NGINX is provided below.  It assumes you have two Mat
+
+The database can be configured for high availability and transparent failover use the existing database technologies. We recommend MySQL Clustering, Postgres Clustering, or Amazon Aoura. Database transparent failover is beyond the scope of this documentation.
+
+Upgrade Guide
+-------------
 
------
+An update is an incremental change to Mattermost server that fixes bugs or performance issues. An upgrade adds new or improved functionality to the server.
+
+Updating Configuration Changes While Operating Continuously
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A service interruption is not required for most configuration updates. See `Upgrading server version while operating continuously`_ for a list of configuration updates that require a service interruption.
+
+You can apply updates during a period of low load, but if your HA cluster is sized correctly, you can do it at any time. The system downtime is brief, and depends on the number of Mattermost servers in your cluster. Note that you are not restarting the machines, only the Mattermost server applications. A Mattermost server restart generally takes about 5 seconds.
+
+1. Make a backup of your existing ``config.json`` file.
+2. For one of the Mattermost servers, make the configuration changes to ``config.json`` and save the file. Do not reload the file yet.
+3. Copy the ``config.json`` file to the other servers.
+4. Shut down Mattermost on all but one server.
+5. Reload the configuration file on the server that is still running. Go to **System Console > Configuration** and click **Reload Configuration from Disk**
+6. Start the other servers. 
+
+Updating Server Version While Operating Continuously
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A service interruption is not required for security patch dot releases of the Mattermost server.
+
+You can apply updates during a period when the anticipated load is small enough that one server can carry the full load of the system during the update.
+
+ Note that you are not restarting the machines, only the Mattermost server applications. A Mattermost server restart generally takes about 5 seconds.
+
+1. Review the upgrade procedure in the *Upgrade Enterprise Edition* section of :doc:`../administration/upgrade`.
+2. Make a backup of your existing ``config.json`` file.
+3. Set your proxy to move all new requests to a single server. If you are using NGINX and it's configured with an upstream backend section in ``/etc/nginx/sites-available/mattermost`` then comment out all but the one server that you intend to update first, and reload NGINX.
+4. Shut down Mattermost on each server except the one that you are updating first.
+5. Update each Mattermost instance that is shut down.
+6. On each server, replace the new ``config.json`` file with your backed-up copy.
+7. Start the Mattermost servers.
+8. Repeat the update procedure for the server that was left running.
+
+Upgrading Server Version While Operating Continuously
```

Comment:
Propose calling this section:

`Server Upgrades Requiring Service Interruption`